target/linux/realtek/files-6.12/drivers/mtd/nand/ecc-realtek.c

```diff
@@ -0,0 +1,346 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Support for Realtek hardware ECC engine in RTL93xx SoCs
+ */
+
+#include <linux/dma-mapping.h>
+#include <linux/mtd/nand.h>
+#include <linux/mutex.h>
+#include <linux/platform_device.h>
+#include <linux/regmap.h>
+
+#define RTL_PAGE_SIZE 		2048
+#define RTL_BLOCK_SIZE 		512
+#define RTL_BLOCKS_PER_PAGE	RTL_PAGE_SIZE / RTL_BLOCK_SIZE
+#define RTL_OOB_SIZE		64
+#define RTL_TAG_SIZE 		6
+#define RTL_SYNDROME_SIZE	10
+#define RTL_SYNDROME_OFFSET	RTL_TAG_SIZE * RTL_BLOCKS_PER_PAGE
+#define RTL_STRENGTH		6
+#define RTL_DMA_SIZE		RTL_BLOCK_SIZE + RTL_TAG_SIZE + RTL_SYNDROME_SIZE
+
+#define RTL_ECC_CFG		0x00
+#define   RTL_ECC_BCH_ALGO	GENMASK(29, 28)
+#define   RTL_ECC_READ_BE	BIT(22)
+#define   RTL_ECC_WRITE_BE	BIT(21)
+#define   RTL_ECC_IRQ_ENABLE	BIT(20)
+#define RTL_ECC_DMA_TIGGER 	0x08
+#define   RTL_ECC_OP_DECODE	0
+#define   RTL_ECC_OP_ENCODE	BIT(0)
+#define RTL_ECC_DMA_START	0x0c
+#define RTL_ECC_DMA_TAG		0x10
+#define RTL_ECC_STATUS		0x14
+#define   RTL_ECC_CORR_COUNT	GENMASK(19, 12)
+#define   RTL_ECC_RESULT	BIT(8)
+#define   RTL_ECC_ALL_ONE	BIT(4)
+#define   RTL_ECC_OP_STATUS	BIT(0)
```

Comment:
what is up with the padding after #define ^^?
shouldn't it always be one space not several
I know nitpicky, but it's easier to read, when all `RTL_` lign up, right?

Reply:
They're fields of the register above.

Reply:
ah I get it now, thanks